core/src/main/scala/kafka/utils/Json.scala

```diff
@@ -69,8 +70,11 @@ object Json {
    * @return An `Either` which in case of `Left` means an exception and `Right` is the actual return value.
    */
   def tryParseFull(input: String): Either[JsonProcessingException, JsonValue] =
-    try Right(mapper.readTree(input)).map(JsonValue(_))
-    catch { case e: JsonProcessingException => Left(e) }
+    if (input != null && input.isEmpty)
+      Left(new JsonParseException(MissingNode.getInstance().traverse(), "The input string shouldn't be empty"))
```

Comment:
It seems like that the most adequate JsonProcessingException is JsonParseException that could apply here. Another possibility is to change the exception in return type to something like IOException and throw that here but it seems it's more vague than throwing JsonParseException.
Another option is to simple throw IllegalArgumentException here instead of Left. That would correspond to the `readTree(InputStream)` as well. What do you think?

Reply:
This seems fine.